esp32_ulp/assemble.py

```diff
@@ -12,9 +12,10 @@
 
 
 class SymbolTable:
-    def __init__(self, symbols, bases):
+    def __init__(self, symbols, bases, globls):
```

Comment:
as long as the method is rather simple/short and does not need to call `globals()` python predefined function, you could also name it `globals`.

Reply:
I didn't want to reuse the python keyword, and since some assemblers apparently used `.globl` as the directive name (https://sourceware.org/binutils/docs/as/Global.html#Global), I chose that spelling.
But now that you point this out, I realise that it's really only in the `__init__` method where the "globals" word would be used, everywhere else it would be `_globals` anyway, which is very obviously not python's built-in function.
I will change it.